include/motor-control/core/motor_hardware_interface.hpp

```diff
@@ -1,13 +1,57 @@
 #pragma once
 
+#include <array>
 #include <atomic>
+#include <cstddef>
 #include <cstdint>
 #include <utility>
+#include <vector>
 
 #include "motor-control/core/types.hpp"
 
 namespace motor_hardware {
 
+static constexpr auto max_requests_per_can_message = 5;
+class UsageEEpromConfig {
+  public:
+    UsageEEpromConfig(const std::vector<UsageRequestSet>& requests) {
```

Comment:
Unfortunately `vector` will allocate internally and we can't use it. For something like this, let's use templated iterator pairs and I think that will do the job.